### PR TITLE
Preview cards

### PR DESCRIPTION
**Pull request by BrianMitchL**
Add OpenGraph and Twitter cards for nice share previews

### CODE REVIEW

File: src/_includes/layouts/svgAsPng.11ty.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const { createHash } = require('crypto');
+const { AssetCache } = require('@11ty/eleventy-cache-assets');
+
+async function render(data) {
+  const hash = createHash('sha256');
+  hash.update(data.site.isProduction.toString() + data.content);
+
+  const asset = new AssetCache(`svgAsPng--${hash.digest('hex')}`);
+
+  if (asset.isCacheValid('30d')) {
```
Comment (zapplebee) [inline src/_includes/layouts/svgAsPng.11ty.js:11]: Looks like you're still checking for days, seems like the hash should indicate it's a good cache indefinitely.

File: src/opengraph-image.11ty.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      },
+      tags: [],
+      layout: 'layouts/svgAsPng',
+      permalink: ({ item }) => {
+        return `${item.url}/opengraph.png`;
+      },
+    };
+  }
+
+  // thanks Zac - https://github.com/zapplebee/personal-blog/blob/5ed063b8a9976a29a6ce990859aae52e52ef7ec3/site/og-image.11ty.js#L24
```
Comment (zapplebee) [inline src/opengraph-image.11ty.js:18]: ❤️

File: src/opengraph-image.11ty.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+class OpenGraphImage {
+  data() {
+    return {
+      eleventyExcludeFromCollections: true,
+      pagination: {
+        data: 'collections.all',
+        size: 1,
+        alias: 'item',
+      },
```
Comment (zapplebee) [inline src/opengraph-image.11ty.js:9]: Eventually, you could use pagination for multiple sizes. But one is a good start.

File: src/_includes/layouts/svgAsPng.11ty.js (hunk 1 above)
Comment (BrianMitchL) [reply src/_includes/layouts/svgAsPng.11ty.js:11]: This is really just an arbitrary length. Long enough to not need to render images most of the time, but still make sure this code path works occasionally too ¯\_(ツ)_/¯

File: src/_includes/head.njk [hunk 4]
```diff
@@ -59,13 +59,8 @@
...
 <meta property="og:url" content="{{ site.url }}{{ page.url }}" />
 <meta property="og:description" content="{{ description or site.description }}" />
-{% if image and imageAlt %}
-<meta property="og:image" content="{{ image | url | absoluteUrl(site.url) }}" />
-<meta property="og:image:alt" content="{{ imageAlt }}" />
-{% else %}
-<meta property="og:image" content="{{ site.author.image }}" />
-<meta property="og:image:alt" content="Profile photo of Brian Mitchell" />
-{% endif %}
+<meta property="og:image" content="{{ site.url }}{{ page.url }}opengraph.png" />
```
Comment (zapplebee) [inline src/_includes/head.njk:62]: I hate .njk...

Comment (BrianMitchL) [reply src/_includes/head.njk:62]: Seemed better than adding react-dom 😉

File: .eleventy.js [hunk 5]
```diff
@@ -94,6 +95,9 @@ module.exports = function (eleventyConfig) {
   });
   eleventyConfig.addPlugin(pluginSchema);
   eleventyConfig.addPlugin(pluginExcerpt);
+  eleventyConfig.addPlugin(pluginImageBase64, {
+    input: path.resolve(__dirname, dir.input),
+  });
 
   eleventyConfig.addLayoutAlias('post', 'layouts/post.njk');
   eleventyConfig.addLayoutAlias('page', 'layouts/page.njk');
```
Comment (zapplebee) [inline .eleventy.js:103]: I dont think you actually need aliases for things that are named same same

File: lib/imageBase64.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
+const fs = require('fs');
+const path = require('path');
+const { createHash } = require('crypto');
+const Cache = require('@11ty/eleventy-cache-assets');
+const { AssetCache } = Cache;
+const sharp = require('sharp');
+
+async function imageBase64(filePath, options) {
+  const hash = createHash('sha256');
+  hash.update(filePath);
```
Comment (zapplebee) [inline lib/imageBase64.js:10]: This would be invalid if you overwrite the same file. Ideally, it should hash the content, rather than the filename

Comment (BrianMitchL) [reply lib/imageBase64.js:10]: Whoooops, good point!

File: src/_includes/head.njk (hunk 4 above)
Comment (ryanmr) [reply src/_includes/head.njk:62]: It's definitely a tradeoff! `njk` is no joke

**Pull request merged**